File: Data Structures & Algorithms/duplicate-integer/submission-0.java

class Solution {
    public boolean hasDuplicate(int[] nums) {
        if (nums == null || nums.length <= 1) return false;
        Set<Integer>st= new HashSet<>(nums.length);

        int n= nums.length;

        for(int i=0; i<n; i++){
            if(!st.add(nums[i])){
                return true;
            }

        }

       
         return false;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
         int n= nums.size();
       sort(nums.begin(), nums.end());
             vector<vector<int>>ans;
             set<tuple<int, int, int>>st;
         for(int idx=0; idx<n-2; idx++){
             int i= idx+1;
              int j= n-1;
              while(i<j){
                 if(nums[idx]+ nums[i]+ nums[j]>0){
                    j--;
                 }
                 else if(nums[idx]+ nums[i]+ nums[j]<0){
                    i++;
                 }
                 else{ 
                    if(!st.count({nums[idx], nums[i], nums[j]})){
                         ans.push_back({nums[idx], nums[i], nums[j]});
                         st.insert({nums[idx], nums[i], nums[j]});
                    }
                   
                    i++; 
                    j--;
                 }
              }
         }
         return ans;
    }
};
